### ISSUE TITLE
Support new branch creation by default

### ISSUE THREAD

**Issue by drewwells** [EndBug/add-and-commit#313]
**Describe the bug**
I'm committing code to a new branch in a remote repo. The local repo has a new branch and the remote one does not yet have this branch.

**Workflow used**
remote> git branch newbranch
remote> git add .
remote> git commit -am {commit message}

**Expected behavior**
The line `> Switching/creating branch...` would create the branch.

**Logs**
Either provide a link to the action run or (if your repo is private) paste here the logs from the step that uses it. If you paste the logs, please use this template, and remember to paste the logs from all the different sections.
<details>
  <summary>Logs</summary>
<pre><code>Internal logs
  > Staging files...
  > Adding files...
  > No files to remove.
  > Checking for uncommitted changes in the git working tree...
  > Found 8 changed files.
  { raw: '', remote: null, branches: [], tags: [] }
  > Switching/creating branch...
  Error: Error: error: pathspec 'aff' did not match any file(s) known to git
  
  > Pulling from remote...
  { raw: '', remote: null, branches: [], tags: [] }
  Error: Error: There is no tracking information for the current branch.
  Please specify which branch you want to merge with.
  See git-pull(1) for details.
  
      git pull <remote> <branch>
  
  If you wish to set tracking information for this branch you can do so with:
  
      git branch --set-upstream-to=origin/<branch> aff
  
  
Outputs
Error: Error: There is no tracking information for the current branch.</code></pre>
</details>

**Comment by drewwells** [comment]
Something removed since it's a private repo, but this should have all the information important for `add-and-commit`
```
name: Generate
on:
  pull_request:
  push:
    branches:
      - main
jobs:
  generate:
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
      matrix:
        lang:
          - go
          - python
    env:
      GEN_LANG_DIR: gen/${{ matrix.lang }}
    steps:
      - uses: actions/checkout@v2
      - if: matrix.lang == 'go'
        run: make buf=buf clients
      - if: matrix.lang != 'go'
        run: buf generate
      - name: diff generated code
        if: always()
        working-directory: ${{ env.GEN_LANG_DIR }}
        run: |
          git add -N .
          git diff --stat
      - name: commit generated code
        if: github.repository == env.upstream && github.ref == 'refs/heads/main'
        uses: EndBug/add-and-commit@v7
        with:
          branch: main
          message: ${{ github.event.head_commit.message }}
          add: .
          cwd: ${{ env.GEN_LANG_DIR }}

```

**Comment by EndBug** [comment]
I've released this feature in `v7.5.0` (also ships with `v7` and `latest`).

Thank you so much for you contribution ❤️ 

@all-contributors please add @drewwells for their idea
